src/overlays/actors/ovl_En_In/z_en_in.c

```diff
@@ -21,92 +44,1623 @@ const ActorInit En_In_InitVars = {
+void EnIn_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnIn* this = THIS;
+    s32 pad[2];
+    s16 paramsFlag;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_In_0x808F30B0/func_808F43E0.asm")
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 30.0f);
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_06014EA8, NULL, this->limbDrawTbl, this->transitionDrawTbl, 20);
+    func_808F30B0(&this->skelAnime, 0);
+    Collider_InitCylinder(globalCtx, &this->colliderCylinder);
+    Collider_SetCylinder(globalCtx, &this->colliderCylinder, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit2);
+    this->unk48A = 0;
+    this->unk48C = 0;
+    this->unk4AC = 0;
+    paramsFlag = this->actor.params & 0x1FF;
+    this->unk4B0 = gSaveContext.weekEventReg[92] & 7;
+    if (paramsFlag == 2 || paramsFlag == 4) {
+        this->unk4AC |= 8;
+    }
+    if (paramsFlag == 1 || paramsFlag == 2) {
+        ActorShape_Init(&this->actor.shape, 0.0f, NULL, 0.0f);
+        this->unk488 = 1;
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_06016A60, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_06016A60.common), 2,
+                             0.0f);
+        Actor_SetScale(&this->actor, 0.01f);
+        this->unk23C = 0;
+        this->unk23D = 1;
+        this->actionFunc = func_808F374C;
+    } else {
+        Collider_InitJntSph(globalCtx, &this->colliderJntSph);
+        Collider_SetJntSph(globalCtx, &this->colliderJntSph, &this->actor, &sJntSphInit, &this->colliderJntSphElement);
+        Actor_SetScale(&this->actor, 0.01f);
+        this->actor.gravity = -4.0f;
+        this->unk240 = func_8013D648(globalCtx, (this->actor.params & 0x7E00) >> 9, 0x3F);
+        this->unk23D = 0;
+        if (paramsFlag == 3 || paramsFlag == 4) {
+            if ((gSaveContext.weekEventReg[92] & 7) == 2 || (gSaveContext.weekEventReg[92] & 7) == 3) {
+                gSaveContext.weekEventReg[56] &= (u8)~8;
+                this->unk4A8 = 0;
+                this->unk4AC |= 2;
+                func_808F35AC(this, globalCtx);
+                this->unk23C = 0;
+                D_801BDAA0 = 0;
+                if ((gSaveContext.weekEventReg[92] & 7) == 2) {
+                    func_808F30B0(&this->skelAnime, 6);
+                } else {
+                    func_808F30B0(&this->skelAnime, 4);
+                }
+                if ((gSaveContext.weekEventReg[92] & 7) == 2) {
+                    this->skelAnime.animCurrentFrame =
+                        ((Rand_ZeroOne() * 0.6f) + 0.2f) * this->skelAnime.animFrameCount;
+                }
+                if (this->unk4AC & 8) {
+                    this->actionFunc = func_808F39DC;
+                } else {
+                    this->actionFunc = func_808F5C98;
+                }
+            } else {
+                if ((gSaveContext.weekEventReg[92] & 7) != 1) {
```

Comment:
```suggestion
                if ((gSaveContext.weekEventReg[92] & 7) != true) {
```

Reply:
I don't feel true is entirely accurate in this case. These same three flags (bits) get read multiple time and don't just compare to 0 and 1. For instance in init we have `(gSaveContext.weekEventReg[92] & 7) == 2 || (gSaveContext.weekEventReg[92] & 7) == 3)`.

Reply:
Oh, I see what's going on here. It's checking multiple bits at once. so 
`
(gSaveContext.weekEventReg[92] & 7) == 2 || (gSaveContext.weekEventReg[92] & 7) == 3)
`
is really
`
(gSaveContext.weekEventReg[92] & (0x1|0x2|0x4) ) == 0x2 || (gSaveContext.weekEventReg[92] & (0x1|0x2|0x4) ) == (0x1|0x2))
`
(you don't really need the `0x`, it just makes it clear that it's checking certain bits).
So I'd still go with:
`if ((gSaveContext.weekEventReg[92] & (1|2|4)) != true) {`
Open to thoughts!

Reply:
No actually you might be right, it should probably be 
`if ((gSaveContext.weekEventReg[92] & (1|2|4)) != 1) {`